client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -462,8 +462,10 @@ static Request count(CountRequest countRequest) throws IOException {
     }
 
     static Request explain(ExplainRequest explainRequest) throws IOException {
-        Request request = new Request(HttpGet.METHOD_NAME,
-            endpoint(explainRequest.index(), explainRequest.type(), explainRequest.id(), "_explain"));
+        String endpoint = explainRequest.isTypeless()
+            ? endpoint(explainRequest.index(), "_explain", explainRequest.id())
+            : endpoint(explainRequest.index(), explainRequest.id(), "_explain");
```

Comment:
I would expect to have `explainRequest.type()` somewhere on this line?

Reply:
Oops, I think I have been staring at this too long...